src/Application/Tools/Algorithm/itkFastGrowCut.h

```diff
@@ -34,47 +34,100 @@
 #include <itkObject.h>
 #include <itkMacro.h>
 
-const unsigned short SrcDimension = 3;
-typedef float DistPixelType; // float type pixel for cost function
-typedef short SrcPixelType;
-typedef unsigned char LabPixelType;
-
-template <class TInputImage>
-class itkFastGrowCut : public itk::Object
+namespace itk
+{
+template <typename TInputImage, typename TLabelImage>
+class ITK_TEMPLATE_EXPORT FastGrowCut : public ImageToImageFilter<TInputImage, TLabelImage>
 {
 public:
-  static itkFastGrowCut* New();
-  itkTypeMacro(itkFastGrowCut,itk::Object);
+  ITK_DISALLOW_COPY_AND_MOVE(FastGrowCut);
+
+  /** Standard class type aliases. */
+  using Self = FastGrowCut;
+  using Superclass = ImageToImageFilter<TInputImage, TLabelImage>;
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+
+  /** Method for creation through the object factory. */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods).  */
+  itkTypeMacro(FastGrowCut, ImageToImageFilter);
+
+  using InputImageType = TInputImage;
+  using InputImagePointer = typename InputImageType::Pointer;
+  using InputImageRegionType = typename InputImageType::RegionType;
+  using InputImagePixelType = typename InputImageType::PixelType;
+  using IndexType = typename InputImageType::IndexType;
+  using SizeType = typename InputImageType::SizeType;
+
+  using LabelPixelType = LabelImageType::PixelType;
+  using LabelImageType = TLabelImage;
+  using LabelImagePointer = typename LabelImageType::Pointer;
+  using LabelImageRegionType = typename LabelImageType::RegionType;
+  using LabelImagePixelType = typename LabelImageType::PixelType;
 
-  //set parameters of grow cut
-  itkSetObjectMacro( SourceVol, TInputImage );
-  itkSetObjectMacro( SeedVol, TInputImage );
+  using SeedsContainerType = std::vector<IndexType>;
 
-  itkSetMacro( InitializationFlag, bool );
+  using InputRealType = typename NumericTraits<InputImagePixelType>::RealType;
+
+  itkSetMacro(InitializationFlag, bool);
 
   //processing functions
   void Initialization();
```

Comment:
I now realize that declarations of `Initialization()` and `RunFGC()` should be removed, as they are not implemented. That doesn't trigger compile errors because they are not invoked anywhere.

Reply:
Okay, thanks!